app/controllers/api/v1/allocations_controller.rb

```diff
@@ -65,6 +71,24 @@ def render_allocation(allocation, status)
       def find_allocation
         Allocation.find(params[:id])
       end
+
+      def from_location
+        Location.find(allocation_params.dig(:relationships, :from_location, :data, :id))
+      end
+
+      def to_location
+        Location.find(allocation_params.dig(:relationships, :to_location, :data, :id))
+      end
+
+      def moves
```

Comment:
a bit on the fence around further refactoring this/moving it to a service, as felt that might be a bit overkill. Thoughts?

Reply:
I think it would make sense if the majority of the create method was also moved into the service - it would be passed allocation_attributes and return an allocation instance, handling all the details of creating associated moves and firing the notifications. I think all of that responsibility belongs elsewhere than the controller - which perhaps it better added to one of the models, or given the conventions with this codebase a simple create allocation service. Resolving location Id's to locations could be done there too removing the need for lines 75-81.

That's the main change I'd suggest with this PR - it might make the service easier to test in isolation rather than having to go via a complete request/response.

Reply:
That's fair, I do agree with moving this out of the controller. My next question is around where this should live, my first thoughts would be to move it into the model, but the established pattern for services (as you've mentioned) in the app eg.`People` live in services under the class `People::Crud` and `People::Creator`, `People::Updater` etc. Are we happy to reuse this pattern for other models like `Allocation`, or move away from it, as Im keen on having this in a model instead. Maybe @martyn-w can weigh in regarding the services vs model approach and thoughts around the pattern and conventions we want to establish going forward?